src/hotspot/share/utilities/compilerWarnings_gcc.hpp

```diff
@@ -28,15 +28,23 @@
 // Macros related to control of compiler warnings.
 
 #ifndef ATTRIBUTE_PRINTF
-#define ATTRIBUTE_PRINTF(fmt,vargs)  __attribute__((format(printf, fmt, vargs)))
+#define ATTRIBUTE_PRINTF(fmt,vargs)  [[gnu::format(printf, fmt, vargs)]]
 #endif
 #ifndef ATTRIBUTE_SCANF
-#define ATTRIBUTE_SCANF(fmt,vargs)  __attribute__((format(scanf, fmt, vargs)))
+#define ATTRIBUTE_SCANF(fmt,vargs)  [[gnu::format(scanf, fmt, vargs)]]
 #endif
 
-#define PRAGMA_DISABLE_GCC_WARNING_AUX(x) _Pragma(#x)
-#define PRAGMA_DISABLE_GCC_WARNING(option_string) \
-  PRAGMA_DISABLE_GCC_WARNING_AUX(GCC diagnostic ignored option_string)
+#if defined(__clang_major__) && \
```

Comment:
Not clear why this was moved ??

Reply:
I'm not sure which one you're referring to, but the PRAGMA_DIAG_PUSH/POP was moved up to the top of the header to match compilerWarnings_visCPP.hpp, and PRAGMA_DISABLE_GCC_WARNING_AUX was moved to macros.hpp as the more general PRAGMA macro, since it's useful for all compilers and not just gcc